vpr/src/route/rr_graph_uxsdcxx_serializer.h

```diff
@@ -1626,39 +1631,84 @@ class RrGraphSerializer final : public uxsd::RrGraphBase<RrGraphContextTypes> {
+                side_mask.set(TOP);
+                side_mask.set(BOTTOM);
+                break;
+            case uxsd::enum_loc_side::TOP_RIGHT_LEFT:
+                side_mask.set(TOP);
+                side_mask.set(RIGHT);
+                side_mask.set(LEFT);
+                break;
+            case uxsd::enum_loc_side::TOP_RIGHT_BOTTOM:
+                side_mask.set(TOP);
+                side_mask.set(RIGHT);
+                side_mask.set(BOTTOM);
+                break;
+            case uxsd::enum_loc_side::TOP_BOTTOM_LEFT:
+                side_mask.set(TOP);
+                side_mask.set(BOTTOM);
+                side_mask.set(LEFT);
                 break;
             case uxsd::enum_loc_side::RIGHT:
-                return RIGHT;
+                side_mask.set(RIGHT);
+                break;
+            case uxsd::enum_loc_side::RIGHT_BOTTOM:
+                side_mask.set(RIGHT);
+                side_mask.set(BOTTOM);
+                break;
+            case uxsd::enum_loc_side::RIGHT_LEFT:
+                side_mask.set(RIGHT);
+                side_mask.set(LEFT);
+                break;
+            case uxsd::enum_loc_side::RIGHT_BOTTOM_LEFT:
+                side_mask.set(RIGHT);
+                side_mask.set(BOTTOM);
+                side_mask.set(LEFT);
                 break;
-            case uxsd::enum_loc_side::TOP:
-                return TOP;
             case uxsd::enum_loc_side::BOTTOM:
-                return BOTTOM;
+                side_mask.set(BOTTOM);
+                break;
+            case uxsd::enum_loc_side::BOTTOM_LEFT:
+                side_mask.set(BOTTOM);
+                side_mask.set(LEFT);
+                break;
+            case uxsd::enum_loc_side::LEFT:
+                side_mask.set(LEFT);
+                break;
+            case uxsd::enum_loc_side::TOP_RIGHT_BOTTOM_LEFT:
+                side_mask.set(TOP);
+                side_mask.set(RIGHT);
+                side_mask.set(BOTTOM);
+                side_mask.set(LEFT);
                 break;
             default:
                 report_error(
                     "Invalid side %d", side);
         }
+        return side_mask;
     }
 
-    uxsd::enum_loc_side to_uxsd_loc_side(e_side side) {
-        switch (side) {
-            case LEFT:
-                return uxsd::enum_loc_side::LEFT;
-            case RIGHT:
-                return uxsd::enum_loc_side::RIGHT;
-            case TOP:
-                return uxsd::enum_loc_side::TOP;
-            case BOTTOM:
-                return uxsd::enum_loc_side::BOTTOM;
-            default:
-                report_error(
-                    "Invalid side %d", side);
+    uxsd::enum_loc_side to_uxsd_loc_side(std::bitset<NUM_SIDES> sides) {
+        if (0 == sides.to_ulong()) {
+            report_error(
+                "Invalid side %ld", sides.to_ulong());
         }
+        return static_cast<uxsd::enum_loc_side>(sides.to_ulong());
```

Comment:
The fact that you are writing out a `static_cast` here should tell you this is not a good strategy.

Reply:
I was trying to save some codes. An explicit translation may work better.